src/analysis/normed_space/deriv.lean

```diff
@@ -17,40 +17,268 @@ is restricted to `s`. We also have
+`unique_diff_on s` express this property, and we prove that indeed they imply the uniqueness of the
+derivative. This is satisfied for open subsets, and in particular for `univ`. This uniqueness
+only holds when the field is non-discrete, which we request at the very beginning:
+otherwise, a derivative can be defined, but it has no interesting properties whatsoever.
+
+In addition to the definition and basic properties of the derivative, this file contains the
+usual formulas (and existence assertions) for the derivative of
+* constants
+* the identity
+* bounded linear maps
+* bounded bilinear maps
+* sum of two functions
+* multiplication of a function by a scalar constant
+* negative of a function
+* subtraction of two functions
+* bounded bilinear maps
+* multiplication of a function by a scalar function
+* multiplication of two scalar functions
+* composition of functions (the chain rule)
+
 -/
 import topology.basic topology.sequences topology.opens
-import analysis.normed_space.operator_norm
+import analysis.normed_space.operator_norm analysis.normed_space.bounded_linear_maps
 
 import analysis.asymptotics
 import tactic.abel
 
-open filter asymptotics bounded_linear_map
+open filter asymptotics bounded_linear_map set
+
+noncomputable theory
+local attribute [instance, priority 0] classical.decidable_inhabited classical.prop_decidable
 
 section
 
-variables {k : Type*} [normed_field k]
+set_option class.instance_max_depth 60
+
+variables {k : Type*} [nondiscrete_normed_field k]
 variables {E : Type*} [normed_space k E]
 variables {F : Type*} [normed_space k F]
 variables {G : Type*} [normed_space k G]
 
-variables (c : k)
-variables (f : E → F) (f' : E →L[k] F)
-variables (x : E) (s: set E) (L : filter E)
-
-def has_fderiv_at_filter :=
+def has_fderiv_at_filter (f : E → F) (f' : E →L[k] F) (x : E) (L : filter E) :=
 is_o (λ x', f x' - f x - f' (x' - x)) (λ x', x' - x) L
 
-def has_fderiv_at_within :=
+def has_fderiv_at_within (f : E → F) (f' : E →L[k] F) (x : E) (s : set E) :=
 has_fderiv_at_filter f f' x (nhds_within x s)
 
-def has_fderiv_at :=
+def has_fderiv_at (f : E → F) (f' : E →L[k] F) (x : E) :=
 has_fderiv_at_filter f f' x (nhds x)
 
-variables {f} {f₀ f₁ g: E → F}
-variables {f'} {f₀' f₁' g': E →L[k] F}
-variables {x}
-variables {s} {t : set E}
-variables {L} {L₁ L₂ : filter E}
+variables (k)
+
+def differentiable_at_within (f : E → F) (x : E) (s : set E) :=
+∃f' : E →L[k] F, has_fderiv_at_within f f' x s
+
+def differentiable_at (f : E → F) (x : E) :=
+∃f' : E →L[k] F, has_fderiv_at f f' x
+
+def fderiv_at_within (f : E → F) (x : E) (s : set E) : E →L[k] F :=
+if h : ∃f', has_fderiv_at_within f f' x s then classical.some h else 0
+
+def fderiv_at (f : E → F) (x : E) : E →L[k] F :=
+if h : ∃f', has_fderiv_at f f' x then classical.some h else 0
+
+def differentiable_on (f : E → F) (s : set E) :=
+  ∀x ∈ s, differentiable_at_within k f x s
```

Comment:
I think the style rule is to indent the type, but not the term.
So indent the new line after `:` but not after `:=`.
```suggestion
∀x ∈ s, differentiable_at_within k f x s
```